app.ino

```diff
@@ -146,50 +147,130 @@ void started() {
   String outstring = "";
   if (Time.hour() < 8 && first) outstring += "Good morning <@UJ67H58GG|gordon>!\n";
   else if (Time.hour() < 10 && first) outstring += "Good morning!\n";
-  outstring += "Brain juice is on it's way! Sit tight! :rocket:";
+  outstring += "Coffee is on it's way! Sit tight! :rocket:";
 
-  Particle.publish("slack", outstring, PUBLIC);
+  proclaimCoffeeInSlack(outstring);
   return;
 }
 
+void proclaimCoffeeInSlack(String message) {
+  Particle.publish("slack", message, PUBLIC);
+}
+
 void done() {
-  bool first = nbrOfCups == 0;
+  bool firstCupsOfTheDay = nbrOfCups == 0;
   nbrOfCups += (int) round(cups);
-  String outstring = "";
 
-  // Always present number of coffee cups served
-  outstring += String((int) round(cups)) + " cups of brain juice served! ";
-
-  // Random emoji means awesome emoji!
-  int r = random(50);
-  if (Time.weekday() == 6) outstring += ":coffee_parrot:";
-  else if (r == 0) outstring += ":aw_yeah:";
-  else if (r == 1) outstring += ":carlton:";
-  else if (r == 2) outstring += ":thumbsup_all:";
-  else if (r == 3) outstring += ":the_horns:";
-  else if (r == 4) outstring += ":raised_hands:";
-  else outstring += ":coffee:";
-  
-  // Give the people what the people want: fun stats!
-  outstring += "\nThat's " + String(nbrOfCups) + " in total today.";
+  String coffeeDoneMessage = coffeeDoneMessageCompiler(firstCupsOfTheDay, nbrOfCups, cups);
+
+  proclaimCoffeeInSlack(coffeeDoneMessage)
+  Particle.publish("dev_slack", coffeeDoneMessage, PUBLIC);
+  return;
+}
+
+String coffeeDoneMessageCompiler(bool firstCupsOfTheDay, int totalNumberOfCups, int servedCups) {
+  return coffeeDoneNumberOfCupsServedCompiler(servedCups) + coffeeDoneStatisticsMessageCompiler(totalNumberOfCups,firstCupsOfTheDay);
+}
+
+String coffeeDoneNumberOfCupsServedCompiler(int servedCups) {
+  String optionalMessage = ""
+
+  if (random(maxRandomNumber) == maxRandomNumber) {
+    message = "EXTERMINAT... \n FORCE SYSTEM REBOOT... \n... \n "
```

Comment:
```suggestion
    optionalMessage = "EXTERMINAT... \n FORCE SYSTEM REBOOT... \n... \n ";
```
not sure if you meant for it to be `optionalMesssage` or something else :upside_down_face: